From 33f4e150a1dd9c4368533fd73689fa43d4051442 Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Fri, 12 Jun 2026 16:56:51 +0300
Subject: [PATCH] mscms: Add stub for `WcsCreateIccProfile()`.

This patch fixes Qt 6.12.0-beta1 GUI apps
not being able to create a window.

(cherry picked from commit 6bf850b5815f3339ee306e8550aa2427f80e5849)
---
 dlls/mscms/mscms.spec |  2 +-
 dlls/mscms/profile.c  | 10 ++++++++++
 include/icm.h         |  1 +
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/dlls/mscms/mscms.spec b/dlls/mscms/mscms.spec
index c15359c133f..5926a8c1bd3 100644
--- a/dlls/mscms/mscms.spec
+++ b/dlls/mscms/mscms.spec
@@ -86,7 +86,7 @@
 @ stdcall UnregisterCMMW(wstr long)
 @ stub WcsAssociateColorProfileWithDevice
 @ stub WcsCheckColors
-@ stub WcsCreateIccProfile
+@ stdcall WcsCreateIccProfile(ptr long)
 @ stub WcsDisassociateColorProfileFromDevice
 @ stub WcsEnumColorProfiles
 @ stdcall WcsEnumColorProfilesSize(long ptr ptr)
diff --git a/dlls/mscms/profile.c b/dlls/mscms/profile.c
index 395fe50dabf..9ab946e34bb 100644
--- a/dlls/mscms/profile.c
+++ b/dlls/mscms/profile.c
@@ -1526,3 +1526,13 @@ HPROFILE WINAPI WcsOpenColorProfileW( PROFILE *cdm, PROFILE *camp, PROFILE *gmmp
 
     return OpenColorProfileW( cdm, access, sharing, creation );
 }
+
+/******************************************************************************
+ * WcsCreateIccProfile                [MSCMS.@]
+ */
+HPROFILE WINAPI WcsCreateIccProfile( HPROFILE profile, DWORD options )
+{
+    FIXME( "%p, %#lx stub!\n", profile, options );
+
+    return NULL;
+}
diff --git a/include/icm.h b/include/icm.h
index bcf79eb63bd..ec6987db663 100644
--- a/include/icm.h
+++ b/include/icm.h
@@ -498,6 +498,7 @@ BOOL       WINAPI UninstallColorProfileW(PCWSTR,PCWSTR,BOOL);
 BOOL       WINAPI UnregisterCMMA(PCSTR,DWORD);
 BOOL       WINAPI UnregisterCMMW(PCWSTR,DWORD);
 #define    UnregisterCMM WINELIB_NAME_AW(UnregisterCMM)
+HPROFILE   WINAPI WcsCreateIccProfile(HPROFILE,DWORD);
 BOOL       WINAPI WcsEnumColorProfilesSize(WCS_PROFILE_MANAGEMENT_SCOPE,ENUMTYPEW*,DWORD*);
 BOOL       WINAPI WcsGetDefaultColorProfile(WCS_PROFILE_MANAGEMENT_SCOPE,const WCHAR*,COLORPROFILETYPE,COLORPROFILESUBTYPE,DWORD,DWORD,LPWSTR);
 BOOL       WINAPI WcsGetDefaultColorProfileSize(WCS_PROFILE_MANAGEMENT_SCOPE,const WCHAR*,COLORPROFILETYPE,COLORPROFILESUBTYPE,DWORD,DWORD*);
